torch/lib/THC/THCTensorMathPointwise.cuh

```diff
@@ -415,11 +415,17 @@ struct TensorDivOp<half> {
 template <typename T>
 struct TensorCRemainderOp {
   __device__ __forceinline__ void operator()(T* out, T* in) {
-    *out = *in != 0 ? *out - *in * (*out / *in) : NAN;
+    *out = (*in != 0) ? *out % *in : NAN;
+    if ((*out * *in)<0){
+      *out += *in;
+    }
   }
 
   __device__ __forceinline__ void operator()(T* out, T* in1, T* in2) {
-    *out = *in2 != 0 ? *in1 - *in2 * (*in1 / *in2) : NAN;
+    *out = (*in2 != 0) ?  *in1 % *in2 : NAN;
```

Comment:
So should the behaviour simply be that we don't check for the "divisor is zero" case and let it do a RuntimeError? I'll do the change tonight when i get home if that is what you want.